Mods to update real-time data on Jet and update pre-defined Alaska domains/GFDLgrid option for GSD 3-km CONUS domain

JeffBeck-NOAA opened the pull request:
Tested on Jet.

gsketefian commented on ush/get_extrn_mdl_file_dir_info.sh:
```diff
@@ -437,40 +450,60 @@ bination of external model (extrn_mdl_name) and analysis or forecast
 
     "GSMGFS")
-      fcst_hhh=( $( printf "%03d " "${lbc_update_fhrs[@]}" ) )
+      fcst_hh=( $( printf "%03d " "${lbc_update_fhrs[@]}" ) )
```
Why change from fcst_hhh to fcst_hh?  fcst_hhh is meant to indicate that it is a 3-digit hour.

JeffBeck-NOAA replied:
Fixed typo.

gsketefian commented on ush/get_extrn_mdl_file_dir_info.sh:
```diff
@@ -388,7 +389,11 @@ fi
 #        fns=( "atm" "sfc" "nst" )
         fns=( "atm" "sfc" )
-        prefix="gfs.t${hh}z."
+        if [ "${MACHINE}" = "JET" ]; then
```
Make indentation consistent -- 2 spaces per indentation level.

JeffBeck-NOAA replied:
Fixed in latest commit

gsketefian commented on ush/get_extrn_mdl_file_dir_info.sh:
```diff
@@ -405,11 +410,19 @@ fi
   
     "RAPX")
+     if [ "${MACHINE}" = "JET" ]; then
+      fns=( "wrfnat_130_${fcst_hh}.grib2" )
```
Fix indentation.

JeffBeck-NOAA replied:
Fixed in latest commit

gsketefian commented on ush/get_extrn_mdl_file_dir_info.sh:
```diff
@@ -405,11 +410,19 @@ fi
 
     "HRRRX")
+     if [ "${MACHINE}" = "JET" ]; then
+      fns=( "wrfnat_hrconus_${fcst_hh}.grib2" )
```
Fix indentation.

JeffBeck-NOAA replied:
Fixed in latest commit

gsketefian commented on ush/get_extrn_mdl_file_dir_info.sh:
```diff
@@ -437,40 +450,60 @@ bination of external model (extrn_mdl_name) and analysis or forecast
-        fcst_hhh=( $( printf "%03d " "${lbc_update_fhrs[@]}" ) )
-        prefix="gfs.t${hh}z.atmf"
-        fns=( "${fcst_hhh[@]/#/$prefix}" )
+        fcst_hh=( $( printf "%03d " "${lbc_update_fhrs[@]}" ) )
```
1) fcst_hh back to fcst_hhh (or maybe I'm missing something).
2) fix indentation

JeffBeck-NOAA replied:
Fixed in latest commit

gsketefian replied:
I can't figure out how to update things to see the changes you made in response to my comments.  So I'll resolve most of these conversations and take a look after everything has been fixed.

gsketefian commented on ush/get_extrn_mdl_file_dir_info.sh:
```diff
@@ -437,40 +450,60 @@ bination of external model (extrn_mdl_name) and analysis or forecast
     "RAPX")
       fcst_hh=( $( printf "%02d " "${lbc_update_fhrs[@]}" ) )
-      prefix="${yy}${ddd}${hh}${mn}"
+      if [ "${MACHINE}" = "JET" ]; then 
```
This indentation is correct (2 spaces per indent level).

gsketefian commented on ush/set_predef_grid_params.sh:
```diff
@@ -80,7 +80,7 @@ case ${PREDEF_GRID_NAME} in
 #-----------------------------------------------------------------------
 #
-"GSD_HAFSV0.A3km")
+"GSL_HAFSV0.A3km")
```
If you change the grid names here, you'll also have to change them in ush/valid_param_vals.sh.

gsketefian commented on ush/set_predef_grid_params.sh:
```diff
@@ -426,19 +426,63 @@ predefined domain:
-predefined domain:
-  PREDEF_GRID_NAME = \"${PREDEF_GRID_NAME}\"
-  GRID_GEN_METHOD = \"${GRID_GEN_METHOD}\""
+    GFDLgrid_LON_T6_CTR=-97.5
```
If this is a copy of mine, just a warning that I didn't set the write-component parameters yet.

gsketefian commented on ush/get_extrn_mdl_file_dir_info.sh:
```diff
@@ -390,9 +390,9 @@ fi
         fns=( "atm" "sfc" )
         if [ "${MACHINE}" = "JET" ]; then
-           prefix="${yy}${ddd}${hh}00.gfs.t${hh}z."
+            prefix="${yy}${ddd}${hh}00.gfs.t${hh}z."
```
I think you want to remove a space, not add one.

gsketefian commented on ush/valid_param_vals.sh:
```diff
@@ -8,25 +8,25 @@ valid_vals_PREDEF_GRID_NAME=( \
-"GSD_HRRR13km" \
-"GSD_HRRR25km" \
-"GSD_RAP13km" )
+"GSL_HAFSV0.A3km" \
```
Did you run any tests yet?  These will also have to be changed in the WE2E test configuration files (and anywhere else that they appear; try doing a grep of the whole repo to see where they show up).

gsketefian commented on ush/valid_param_vals.sh:
```diff
@@ -8,25 +8,25 @@ valid_vals_PREDEF_GRID_NAME=( \
 "FV3_GFS_2017_gfdlmp" \
-"FV3_GSD_v0" \
-"FV3_GSD_SAR" \
+"FV3_GSL_v0" \
```
If you change these here, you have to make sure they have also been renamed in the ufs_weather_model repo; otherwise the suite defintion file won't be found.  You'll also have to change these in the WE2E config files, and you'll have to change the names of the corresponding input.nml, diag_table, field_table, and model_config template files.

JeffBeck-NOAA commented:
I fixed the indentation problems in the latest commit, and reverted everything back to the GSD acronym.  Might as well wait until CCPP fixes the acronym names before we break stuff.
